chain/network/src/peer_manager/connection_store/tests.rs

```diff
@@ -115,14 +105,37 @@ fn test_evict_longest_disconnected() {
     connection_store.insert_outbound_connections(conn_infos.clone());
 
     tracing::debug!(target:"test", "insert a new connection");
-    let conn = ConnectionInfo {
-        peer_info: gen_peer_info(OUTBOUND_CONNECTIONS_CACHE_SIZE.try_into().unwrap()),
-        time_established: now_utc - time::Duration::hours(1),
-        time_connected_until: now_utc,
-    };
+    let conn = make_connection_info(rng, now_utc);
     connection_store.insert_outbound_connections(vec![conn.clone()]);
     conn_infos.push(conn);
 
     tracing::debug!(target:"test", "check that the store contains exactly the expected connections");
-    check_outbound_connections(&connection_store, &conn_infos);
+    assert_eq!(connection_store.get_recent_outbound_connections().as_set(), conn_infos.as_set());
+}
+
+#[test]
+fn test_recovery_from_clock_skew() {
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let clock = time::FakeClock::default();
+    let (_tmp_dir, opener) = NodeStorage::test_opener();
+    let store = store::Store::from(opener.open().unwrap());
+    let connection_store = ConnectionStore::new(store).unwrap();
+
+    tracing::debug!(target:"test", "create and store live connections up to the ConnectionStore limit");
+    let now_utc = clock.now_utc();
+    connection_store.insert_outbound_connections(make_connection_infos(
+        rng,
+        now_utc,
+        OUTBOUND_CONNECTIONS_CACHE_SIZE,
+    ));
+
+    tracing::debug!(target:"test", "turn back the clock 1 year");
+    clock.advance(time::Duration::days(-365));
```

Comment:
Advancing a negative amount of time is a noop:
https://cs.github.com/near/nearcore/blob/c308df157bf64a528033b618b4f444d3b9c73f94/chain/network/src/time.rs?q=time.rs#L128

You should use set_utc() instead. While you are at it, please make the fake clock panic instead of noop, so that it is harder to make that kind of mistake.